src/com/open/abyss/leaf/bankopened/WithdrawEquipRing.kt

```diff
@@ -13,12 +13,12 @@ import org.powbot.api.script.tree.Leaf
 class WithdrawEquipRing(script: Script) : Leaf<Script>(script, "Withdrawing Ring of dueling") {
 
     override fun execute() {
-        if (Inventory.stream().name("Ring of Dueling(8)").isEmpty()) {
-            Bank.mustWithdrawItem(Constants.ITEM_RING_OF_DUELING, 1)
-        } else if (Equipment.stream().name("Ring of Dueling(8)").isEmpty()){
-            val ring = Inventory.stream().name(Constants.ITEM_RING_OF_DUELING).first()
+        if (Inventory.stream().id(*ID_RING_OF_DUELING).isEmpty()) {
+            Bank.mustWithdrawItem(ID_RING_OF_DUELING,1)
+        } else if (Equipment.stream().id(*ID_RING_OF_DUELING).isEmpty()){
+            val ring = Inventory.stream().id(*ID_RING_OF_DUELING).first()
             if (ring.interact("Wear")) {
-                Condition.wait { Equipment.stream().name("Ring of Dueling(8)").isNotEmpty() }
+                Condition.wait { Equipment.stream().id(*ID_RING_OF_DUELING).isNotEmpty() }
```

Comment:
When development fix is merged to stable we change this to look at the specific slot.

Reply:
Yeah, didn't change it yet due to stable not having this change